src/go/rpk/pkg/cli/cluster/partitions/move.go

```diff
@@ -0,0 +1,176 @@
+			core, err := strconv.Atoi(nc[1])
+			if err != nil {
+				return "", nil, fmt.Errorf("invalid format; %s in %s", nc[1], replicaCore)
+			}
+			replicas = append(replicas, adminapi.Replica{
+				NodeID: nodeID,
+				Core:   core,
+			})
+		} else {
+			nodeID, err := strconv.Atoi(replicaCore)
+			if err != nil {
+				return "", nil, fmt.Errorf("invalid format; %s in %s", nc[0], replicaCore)
+			}
+			replicas = append(replicas, adminapi.Replica{
+				NodeID: nodeID,
+			})
+		}
+	}
+	return pr[0], replicas, nil
+}
+
+// Expected input format is 'foo/0:1,2,3' or 'redpanda/controller/0:1-0,2-0,3-0'
+// If 'foo/0:1,2,3' is passed, returns
+//
+// - 'kafka/foo/0'
+// - [{NodeID:1,},{NodeID:2},{NodeID:3}]
+//
+// If 'redpanda/controller/0:1-0,2-0,3-0' is passed, returns
+//
+// - 'redpanda/controller/0'
+// - [{NodeID:1,Core:0},{NodeID:2,Core:0},{NodeID:3,Core:0}].
+func parseTopicPartitionReplicas(partition string) (string, []adminapi.Replica, error) {
+	var replicas []adminapi.Replica
+	tpr := strings.Split(partition, ":")
+	if len(tpr) != 2 {
+		return "", nil, fmt.Errorf("invalid format for %s", partition)
+	}
+	if !strings.Contains(tpr[0], "/") {
+		return "", nil, fmt.Errorf("invalid format for %s; specify the topic name", tpr[0])
+	}
+	if strings.Count(tpr[0], "/") == 1 {
+		tpr[0] = "kafka/" + tpr[0]
+	}
+	replicasTmp := strings.Split(tpr[1], ",")
+	for _, replicaCore := range replicasTmp {
+		nc := strings.Split(replicaCore, "-")
+		if len(nc) > 1 {
+			nodeID, err := strconv.Atoi(nc[0])
+			if err != nil {
+				return "", nil, fmt.Errorf("invalid format; %s in %s", nc[0], replicaCore)
+			}
+			core, err := strconv.Atoi(nc[1])
+			if err != nil {
+				return "", nil, fmt.Errorf("invalid format; %s in %s", nc[1], replicaCore)
+			}
+			replicas = append(replicas, adminapi.Replica{
+				NodeID: nodeID,
+				Core:   core,
+			})
+		} else {
+			nodeID, err := strconv.Atoi(replicaCore)
+			if err != nil {
+				return "", nil, fmt.Errorf("invalid format; %s in %s", nc[0], replicaCore)
+			}
+			replicas = append(replicas, adminapi.Replica{
+				NodeID: nodeID,
+			})
+		}
+	}
+	return tpr[0], replicas, nil
+}
+
+const helpAlterAssignments = `Move partition replicas across nodes / cores".
+
+This command changes replica assignments for given partitions. By default,
+this command assumes the "kafka" namespace, but you can use a "{namespace}/"
+to specify internal namespaces.
+
+    rpk cluster partitions move foo --partition 0:1,2,3 -p 1:2,3,4
+
+where the command assigns new replicas to [1,2,3] for partition 0 and
+to [2,3,4] for partition 1 for topic "foo".
+
+You can also specify the core id with "-{core_id}" where the new replicas should be on:
+
+    rpk cluster partitions move foo -p 0:1-0,2-0,3-0
+
+where all new replicas [1,2,3] will be assigned on core 0 on the nodes.
+
+If topic(s) isn't set, specify the topic name in the "--partition / -p" option:
+
+    rpk cluster partitions move -p foo/0:1,2,3 -p kafka_internal/tx/0:1-0,2-0,3-0
```

Comment:
```
This command changes replica assignments for given partitions. By default,
it assumes the "kafka" namespace, but you can specify an internal
namespace using the "{namespace}/" prefix.

To move replicas, use the following syntax:

    rpk cluster partitions move foo --partition 0:1,2,3 -p 1:2,3,4

Here, the command assigns new replicas for partition 0 to brokers [1, 2,
3] and for partition 1 to brokers [2, 3, 4] for the topic "foo."

You can also specify the core id with "-{core_id}" where the new replicas 
should be placed:

    rpk cluster partitions move foo -p 0:1-0,2-0,3-0

Here all new replicas [1,2,3] will be assigned on core 0 on the nodes.

If topic(s) is not set, specify the topic name in the "--partition / -p" option:

    rpk cluster partitions move -p foo/0:1,2,3 -p kafka_internal/tx/0:1-0,2-0,3-0
```

wdyt of this example? I added a sentence to denote that this is an example and a few changes (like isn't -> is not since `'` can be problematic in some shells)

 also cc: @twmb

Reply:
Fixed in https://github.com/redpanda-data/redpanda/commit/1595272ed458ac47621cdbcad4c550c99140d622